OPCB/functions/findSpot.sqf

```diff
@@ -1,32 +1,31 @@
-params ["_radius"];
+params ["_radius", ["_basePos", markerPos "base_marker"], ["_maxRange", 0]];
 private _taskSpot = [0,0,0];
 private _list = [];
-if (_radius < 500) then {
-	
-	private _basePos = markerPos "base_marker";
-	private _suitableSpots = [];
-	
-	private ["_step","_prevStep","_spot","_heading", "_suitable"];
-	
-	private _iter = ((ceil (worldSize/500)) - 5) max 10;
+private _suitableSpots = [];
+private ["_step","_prevStep","_spot","_heading", "_suitable"];
 
-	private _worldHalf = worldSize /2;
-	
-	for "_i" from 1 to _iter do {
-		
-		_suitable = [0,0,0];	
+private _iter = ((ceil (worldSize/500)) - 5) max 10;
+private _minDistance = 2000;
 
-		while {count _suitable == 3 || {surfaceIsWater _suitable}} do {
-			_rndPos = _basePos getPos[random [2000,_worldHalf/2, _worldHalf] ,random 360];
-			_suitable = [_rndPos, 0, _worldHalf, 1, 0, 0.3, 0, ["base_marker"]] call BIS_fnc_findSafePos;
-		};
-		
-		_suitableSpots pushBack _suitable;
+if(_maxRange == 0) then {
+	_maxRange = worldSize / 2;
+	_minDistance = 100;
+};
+
+for "_i" from 1 to _iter do {
 	
+	_suitable = [0,0,0];	
+	while {count _suitable == 3 || {surfaceIsWater _suitable}} do {
```

Comment:
Yes, yes it totally can. 
Proof: Discord picture sent by alex where spawn positions is water. 

Even this 2nd condition is not fucking foolproof against arma bullshit.

Reply:
What in hell's name... could be a map issue then. Sahrani is an Arma 1 map after all... In this case, this condition might still be redundant because I'd bet the BIS fnc internally uses the same command anyway. Maybe there's some wonky workaround we can find like checking for posASL vs posATL or some other bs to find a better solution...